File: leetcode/problems/remove-element/v1.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
  let index = 0;
  for (let j = 0; j < nums.length; j++) {
    if (nums[j] !== val) {
      nums[index++] = nums[j];
    }
  }
  return index;
};

let ans = removeElement([0, 2, 1, 3, 2, 4], 2);
console.log(ans);

File: leetcode/problems/remove-element/v1.py
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        n = len(nums)
        i = 0
        for j in range(n):
            if nums[j] != val:
                nums[i] = nums[j]
                i += 1

        return i


solution = Solution()
res = solution.removeElement([0, 1, 2, 3, 4, 2, 6, 2, 2], 2)
print(res)
